File: runs/A/bmad/03/code/seed-a-frontend/src/app/shared/models/buchung.ts
export interface BuchungAnlegenRequest {
  name: string;
  email: string;
  positionen: BuchungspositionRequest[];
}

export interface BuchungspositionRequest {
  sitzplatzCode: string;
  preiskategorieId: string;
}

export interface Buchung {
  referenz: string;
  veranstaltungId: string;
  name: string;
  email: string;
  status: 'Aktiv' | 'Storniert';
  gesamtpreis: number;
  positionen: Buchungsposition[];
}

export interface Buchungsposition {
  sitzplatzCode: string;
  preiskategorieId: string;
  preiskategorieName: string;
  preisSnapshot: number;
}

File: runs/A/bmad/03/code/seed-a-backend.Api/Models/Buchung.cs
namespace seed_a_backend.Api.Models;

public enum BuchungStatus
{
    Aktiv,
    Storniert,
}

public class Buchung
{
    public required string Id { get; set; }

    public required string Referenz { get; set; }

    public required string VeranstaltungId { get; set; }

    public Veranstaltung? Veranstaltung { get; set; }

    public required string Name { get; set; }

    public required string Email { get; set; }

    public required BuchungStatus Status { get; set; }

    public required decimal Gesamtpreis { get; set; }

    public ICollection<Buchungsposition> Buchungspositionen { get; set; } = new List<Buchungsposition>();
}
